public/app/plugins/datasource/elasticsearch/metric_agg.js

```diff
@@ -13,14 +13,21 @@ function (angular, _, queryDef) {
 
     $scope.metricAggTypes = queryDef.metricAggTypes;
     $scope.extendedStats = queryDef.extendedStats;
+    $scope.mavgOptions = [];
 
     $scope.init = function() {
       $scope.agg = metricAggs[$scope.index];
       $scope.validateModel();
+      $scope.updateMavgOptions();
+    };
+
+    $scope.updateMavgOptions = function() {
+      $scope.mavgOptions = queryDef.getMovingAverageOptions($scope.target);
```

Comment:
this should be named something more generic since I think we can reuse this for all pipeline aggregations (like derivative)